productivity-suite/app/fragments/news/src/root.tsx

```diff
@@ -0,0 +1,58 @@
+// @refresh reload
+import { Suspense } from "solid-js";
+import { HydrationScript, isServer } from "solid-js/web";
+import { ErrorBoundary, Routes, useServerContext, Route } from "solid-start";
+import { PageEvent } from "solid-start/server";
+import { createContext, useContext } from "solid-js";
+import "./root.css";
+import { News } from "./routes";
+
+export const ServerContext = createContext<PageEvent>({} as any);
+
+type ManifestEntry = {
+  type: "script" | "style";
+  href: string;
+};
+
+export default function Root() {
+  const context = useServerContext();
+
+  let script: ManifestEntry = null;
+  let style: ManifestEntry = null;
+
+  if (isServer) {
+    const manifestEntries = context.env.manifest[
+      "entry-client"
+    ] as ManifestEntry[];
+    script = manifestEntries.find((e) => e.type === "script");
+    style = manifestEntries.find((e) => e.type === "style");
```

Comment:
what does this stuff do? is it something solid forces you to do?

Reply:
Nope, this is a hack. Solid normally handles this for you by automatically injecting the entry stylesheet and bundle when you use the `Head` and `Scripts` elements from `solid-start`. However, base path can't be provided as a param to either, so it doesn't work with fragments since we need to prepend `/_fragment/news`. 

Since the entry bundles are hashed, we have to lookup the dynamic filename via the manifest, which as passed down through context.

Reply:
😬 

could there be more then one script and style?

Reply:
In this particular example, no

Reply:
but in general yes?

I am asking because I think that in general we should not include things that can't handle generic situations and/or we can see a path forward for it

Reply:
Solid start implements it like this, so I think we're ok: 

`isSSR ? (
            <script type="module" async src={context.env.manifest["entry-client"][0].href} />`

Reply:
ah 😮, ok thanks 🙂